fili-core/src/main/java/com/yahoo/bard/webservice/sql/evaluator/FilterEvaluator.java

```diff
@@ -0,0 +1,281 @@
+
+/**
+ * Evaluates filters to find all the dimensions used in them and
+ * to build a {@link RexNode} with an equivalent sql filter.
+ */
+public class FilterEvaluator {
+    /**
+     * Private constructor - all methods static.
+     */
+    private FilterEvaluator() {
+
+    }
+
+    /**
+     * Finds all the dimension names used in the filter.
+     *
+     * @param builder  The RelBuilder used with Calcite to make queries.
+     * @param filter  The filter to be evaluated.
+     *
+     * @return a list of all the dimension names.
+     */
+    public static List<String> getDimensionNames(RelBuilder builder, Filter filter) {
+        // todo could use DataApiRequest instead and simplify this class
+        return evaluate(builder, filter).getRight()
+                .stream()
+                .distinct()
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Builds a {@link RexNode} with an equivalent sql filter as the one given.
+     *
+     * @param builder  The RelBuilder used with Calcite to make queries.
+     * @param filter  The filter to be evaluated.
+     *
+     * @return the rexNode containing the filter information.
+     */
+    public static Optional<RexNode> getFilterAsRexNode(RelBuilder builder, Filter filter) {
+        return evaluate(builder, filter).getLeft();
+    }
+
+    /**
+     * Evaluates and builds a filter and finds all the dimension names used in all filters.
+     * NOTE: this is doing two things at once, but the interface given only lets you do one at a time.
+     * This is because dimension names needed to build the RexNode aren't known beforehand.
+     * This forces the flow to be [getDimensionNames] to [buider.project] to [getFilterAsRexNode] to [builder.filter]
+     *
+     * @param builder  The RelBuilder used with Calcite to make queries.
+     * @param filter  The filter to be evaluated.
+     *
+     * @return both the filter and the list of dimensions used in the filter.
+     */
+    private static Pair<Optional<RexNode>, List<String>> evaluate(RelBuilder builder, Filter filter) {
+        List<String> dimensions = new ArrayList<>();
+        Optional<RexNode> rexNode = Optional.ofNullable(evaluate(builder, filter, dimensions));
+        dimensions = dimensions.stream().distinct().collect(Collectors.toList());
+        return Pair.of(rexNode, dimensions);
+    }
+
+    /**
+     * Top level evaluate call which finds the specialized evaluation for a given filter.
+     *
+     * @param builder  The RelBuilder used with Calcite to make queries.
+     * @param filter  The filter to be evaluated.
+     * @param dimensions  The list of dimensions already found.
+     *
+     * @return a RexNode containing an equivalent filter to the one given.
+     */
+    private static RexNode evaluate(RelBuilder builder, Filter filter, List<String> dimensions) {
+        if (filter == null) {
+            return null;
+        }
+        return DispatchUtils.dispatch(
```

Comment:
Why are we doing this? Why not just build (take as an argument) an instance of `FilterEvaluator` and invoke the method directly?